snowfall/training/graph.py

```diff
@@ -0,0 +1,26 @@
+from functools import lru_cache
+from typing import Iterable
+
+import k2
+from k2 import Fsa, SymbolTable
+
+
+class TrainingGraphCompiler:
+    def __init__(self, L: Fsa, vocab: SymbolTable, oov: str = '<UNK>'):
+        self.L = L
+        self.vocab = vocab
+        self.oov = oov
+
+    def compile(self, texts: Iterable[str]) -> Fsa:
+        decoding_graphs = k2.create_fsa_vec([self.compile_one_and_cache(text) for text in texts])
+        decoding_graphs.scores.requires_grad_(False)
```

Comment:
I would suggest 
```python
decoding_graphs.requires_grad_(False)
```

It's simpler or just remove this statement since it requires no grad by default.

Reply:
I'll apply the suggestion and keep it - just in case the default ever changes.